apps/studio/components/ui/FileExplorerAndEditor/FileExplorerAndEditor.tsx

```diff
@@ -0,0 +1,191 @@
+    const updatedFiles = files.map((file) => (file.id === id ? { ...file, name: newName } : file))
+    onFilesChange(updatedFiles)
+  }
+
+  const handleDoubleClick = (id: number, currentName: string) => {
+    const newName = prompt('Enter new file name:', currentName)
+    if (newName && newName !== currentName) {
+      handleFileNameChange(id, newName)
+    }
+  }
+
+  const handleFileSelect = (id: number) => {
+    const updatedFiles = files.map((file) => ({
+      ...file,
+      selected: file.id === id,
+    }))
+    onFilesChange(updatedFiles)
+  }
+
+  const treeData = {
+    name: '',
+    children: files.map((file) => ({
+      id: file.id.toString(),
+      name: file.name,
+      metadata: {
+        isEditing: false,
+        originalId: file.id,
+      },
+    })),
+  }
+
+  return (
+    <div className="flex-1 overflow-hidden flex h-full">
+      <div className="w-64 border-r bg-surface-200 flex flex-col">
+        <div className="py-4 px-6 border-b flex items-center justify-between">
+          <h3 className="text-sm font-medium">Files</h3>
+          <Button size="tiny" type="default" icon={<Plus size={14} />} onClick={addNewFile}>
+            Add File
+          </Button>
+        </div>
+        <div className="flex-1 overflow-y-auto p-2">
+          <TreeView
+            data={flattenTree(treeData)}
+            aria-label="files tree"
+            nodeRenderer={({ element, isBranch, isExpanded, getNodeProps, level }) => {
+              const nodeProps = getNodeProps()
+              const originalId =
+                typeof element.metadata?.originalId === 'number'
+                  ? element.metadata.originalId
+                  : null
+
+              return (
+                <div
+                  onDoubleClick={(e) => {
+                    e.stopPropagation()
+                    if (originalId !== null) handleDoubleClick(originalId, element.name)
+                  }}
+                >
+                  <TreeViewItem
+                    {...nodeProps}
+                    isExpanded={isExpanded}
+                    isBranch={isBranch}
+                    isSelected={files.find((f) => f.id === originalId)?.selected}
+                    level={level}
+                    xPadding={16}
+                    name={element.name}
+                    icon={<File size={14} className="text-foreground-light" />}
+                    isEditing={Boolean(element.metadata?.isEditing)}
+                    onEditSubmit={(value) => {
+                      if (originalId !== null) handleFileNameChange(originalId, value)
+                    }}
+                    onClick={() => {
+                      if (originalId !== null) handleFileSelect(originalId)
+                    }}
+                  />
+                </div>
+              )
+            }}
+          />
+        </div>
+      </div>
+      <div className="flex-1 min-h-0 relative px-3 bg-surface-200">
+        <AIEditor
```

Comment:
hmm is the inline assistant hooked up to work with edge functions atm?

![image](https://github.com/user-attachments/assets/17cf9f65-8143-4277-b20c-9279e617a6d5)

Reply:
actually, is that a legit function if you're just logging? If you type "write me a demo edge function that just returns hello world" it should write a function

Reply:
ahhh okay yeah that prompt works - might still need some tweaking but we'll learn more when users start using
![image](https://github.com/user-attachments/assets/825d3409-6f0d-4522-873d-827bcec662f6)